windows-gaming-driver/src/setup/usb.rs

```diff
@@ -103,51 +203,42 @@ fn pick(special: Option<HidKind>,
     let mut devs = Vec::new();
 
     for dev in iter.scan_devices()? {
-        let dev = if special.is_some() {
+        let dev = if kind.is_some() {
             dev.parent().unwrap()
         } else {
             dev
         };
 
-        let mut vendor = None;
-        let mut product = None;
-        let mut vendor_name = "Unknown vendor".to_owned();
-        let mut product_name = "Unknown product".to_owned();
-
-        for attr in dev.properties() {
-            if let Some(val) = attr.value().to_str() {
-                if attr.name() == "ID_VENDOR_ID" {
-                    vendor = parse_hex(val);
-                } else if attr.name() == "ID_MODEL_ID" {
-                    product = parse_hex(val);
-                } else if attr.name() == "ID_VENDOR_FROM_DATABASE" {
-                    vendor_name = val.to_owned();
-                } else if attr.name() == "ID_MODEL_FROM_DATABASE" {
-                    product_name = val.to_owned();
-                }
-            }
-        }
+        let vendor = dev.property_value("ID_VENDOR_ID").map(|s| s.to_str().unwrap()).and_then(parse_hex);
+        let product = dev.property_value("ID_MODEL_ID").map(|s| s.to_str().unwrap()).and_then(parse_hex);
+        let vendor_name = dev.property_value("ID_VENDOR_FROM_DATABASE").map(|s| s.to_string_lossy().into_owned());
+        let product_name = dev.property_value("ID_MODEL_FROM_DATABASE").map(|s| s.to_string_lossy().into_owned());
+        let busnum = dev.attribute_value("busnum").map(|s| s.to_str().unwrap()).and_then(parse_hex);
+        let devpath = dev.attribute_value("devpath").map(|s| s.to_str().unwrap()).and_then(parse_hex);
```

Comment:
Actually, since we don't use it for much/anything else, you could move the `to_str().unwrap()` into `parse_hex` (and adjust `pci_device` accordingly, obviously).